boa/src/builtins/object/mod.rs

```diff
@@ -109,6 +121,110 @@ impl Object {
         Ok(obj)
     }
 
+    /// `Object.getOwnPropertyDescriptor( object, property )`
+    ///
+    /// Returns an object describing the configuration of a specific property on a given object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://www.ecma-international.org/ecma-262/10.0/index.html#sec-object.getownpropertydescriptor
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptor
+    pub fn get_own_property_descriptor(
+        _: &Value,
+        args: &[Value],
+        ctx: &mut Context,
+    ) -> Result<Value> {
+        let object = args.get(0).unwrap_or(&Value::undefined()).to_object(ctx)?;
+        if let Some(key) = args.get(1) {
+            let key = key.to_property_key(ctx)?;
+
+            if let Some(desc) = object.borrow().get_own_property(&key) {
+                return Ok(Self::from_property_descriptor(desc, ctx)?);
+            }
+        }
+
+        Ok(Value::undefined())
+    }
+
+    /// `Object.getOwnPropertyDescriptors( object )`
+    ///
+    /// Returns all own property descriptors of a given object.
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://www.ecma-international.org/ecma-262/10.0/index.html#sec-object.getownpropertydescriptor
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyDescriptors
+    pub fn get_own_property_descriptors(
+        _: &Value,
+        args: &[Value],
+        ctx: &mut Context,
+    ) -> Result<Value> {
+        let object = args.get(0).unwrap_or(&Value::undefined()).to_object(ctx)?;
+        let descriptors = ctx.construct_object();
+
+        for key in object.borrow().keys() {
+            let descriptor = {
+                let desc = object
+                    .borrow()
+                    .get_own_property(&key)
+                    .expect("Expected property to be on object.");
+                Self::from_property_descriptor(desc, ctx)?
+            };
+
+            if !descriptor.is_undefined() {
+                descriptors.borrow_mut().insert(
+                    key,
+                    PropertyDescriptor::from(DataDescriptor::new(descriptor, Attribute::all())),
+                );
+            }
+        }
+
+        Ok(Value::Object(descriptors))
+    }
+
+    /// https://www.ecma-international.org/ecma-262/10.0/index.html#sec-frompropertydescriptor
```

Comment:
I would prefer a better documentation of this function, similar to the functions above. I would also use the link to the rolling spec. For example:
```suggestion
    /// The abstract operation `FromPropertyDescriptor`.
    ///
    /// [ECMAScript reference][spec]
    /// [spec]: https://tc39.es/ecma262/#sec-frompropertydescriptor
```